include/counter.h

```diff
@@ -94,9 +109,8 @@ typedef u32_t (*counter_api_read)(struct device *dev);
 typedef int (*counter_api_set_alarm)(struct device *dev, u8_t chan_id,
 				const struct counter_alarm_cfg *alarm_cfg);
 typedef int (*counter_api_cancel_alarm)(struct device *dev, u8_t chan_id);
-typedef int (*counter_api_set_top_value)(struct device *dev, u32_t ticks,
-				    counter_top_callback_t callback,
-				    void *user_data);
+typedef int (*counter_api_set_top_value)(struct device *dev,
+					 struct counter_top_cfg *cfg);
```

Comment:
This should be `const struct counter_top_cfg *cfg`. Drivers are not supposed to change the contents of this structure, right?